validation/src/test/scala/hmda/validation/rules/lar/macro/MacroSpec.scala

```diff
@@ -27,11 +27,32 @@ abstract class MacroSpec extends SummaryEditCheckSpec with BeforeAndAfterAll wit
     system.terminate()
   }
 
-  protected def newLarSource(lars: List[LoanApplicationRegister], numOfRelevantLars: Int, relevantDef: LoanApplicationRegister => LoanApplicationRegister, irrelevantDef: LoanApplicationRegister => LoanApplicationRegister) = {
-    val relevantLars = lars.map(lar => relevantDef(lar)).take(numOfRelevantLars)
-    val irrelevantLars = lars.map(lar => irrelevantDef(lar)).drop(numOfRelevantLars)
+  protected def newLarSource(
+    lars: List[LoanApplicationRegister],
+    numOfRelevantLars: Int,
+    relevantDef: LoanApplicationRegister => LoanApplicationRegister,
+    irrelevantDef: LoanApplicationRegister => LoanApplicationRegister,
+    relevantChange: LoanApplicationRegister => LoanApplicationRegister = lar => lar,
+    irrelevantChange: LoanApplicationRegister => LoanApplicationRegister = lar => lar
+  ) = {
+    val relevantLars = lars
+      .map(lar => relevantDef(lar))
+      .map(lar => relevantChange(lar))
+      .take(numOfRelevantLars)
+    val irrelevantLars = lars
+      .map(lar => irrelevantDef(lar))
+      .map(lar => irrelevantChange(lar))
+      .drop(numOfRelevantLars)
     val newLars = relevantLars ::: irrelevantLars
     Source.fromIterator(() => newLars.toIterator)
   }
 
+  protected def setRelevantAmount(base: Int, numOfRelevantLars: Int, numOfLars: Int, findAmount: (LoanApplicationRegister, Int) => LoanApplicationRegister, relevantMultiplier: Double)(lar: LoanApplicationRegister): LoanApplicationRegister = {
```

Comment:
I don't quite understand the value of this method. It seems to be doing two different jobs-- calculating what value each lar needs to have, then setting that value. Since it has lots of parameters and only 2 lines, things might be easier to read and understand if we don't make an abstraction out of it. If we do keep it, it will need a more accurate name, like `calculateAndSetValueOnRelevantLars`